server/src/main/java/org/elasticsearch/action/admin/indices/rollover/RolloverConfiguration.java

```diff
@@ -0,0 +1,420 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.action.admin.indices.rollover;
+
+import org.elasticsearch.common.Strings;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.io.stream.Writeable;
+import org.elasticsearch.common.settings.SettingsException;
+import org.elasticsearch.common.unit.ByteSizeValue;
+import org.elasticsearch.core.Nullable;
+import org.elasticsearch.core.TimeValue;
+import org.elasticsearch.xcontent.ObjectParser;
+import org.elasticsearch.xcontent.ToXContentObject;
+import org.elasticsearch.xcontent.XContentBuilder;
+import org.elasticsearch.xcontent.XContentParser;
+
+import java.io.IOException;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Objects;
+import java.util.Set;
+
+import static org.elasticsearch.action.admin.indices.rollover.RolloverConditions.MAX_AGE_FIELD;
+import static org.elasticsearch.action.admin.indices.rollover.RolloverConditions.MAX_DOCS_FIELD;
+import static org.elasticsearch.action.admin.indices.rollover.RolloverConditions.MAX_PRIMARY_SHARD_DOCS_FIELD;
+import static org.elasticsearch.action.admin.indices.rollover.RolloverConditions.MAX_PRIMARY_SHARD_SIZE_FIELD;
+import static org.elasticsearch.action.admin.indices.rollover.RolloverConditions.MAX_SIZE_FIELD;
+import static org.elasticsearch.action.admin.indices.rollover.RolloverConditions.MIN_AGE_FIELD;
+import static org.elasticsearch.action.admin.indices.rollover.RolloverConditions.MIN_DOCS_FIELD;
+import static org.elasticsearch.action.admin.indices.rollover.RolloverConditions.MIN_PRIMARY_SHARD_DOCS_FIELD;
+import static org.elasticsearch.action.admin.indices.rollover.RolloverConditions.MIN_PRIMARY_SHARD_SIZE_FIELD;
+import static org.elasticsearch.action.admin.indices.rollover.RolloverConditions.MIN_SIZE_FIELD;
+
+/**
+ * This class holds the configuration of the rollover conditions as they are defined in DLM lifecycle. Currently, it can handle automatic
+ * configuration for the max index age condition.
```

Comment:
Should we validate this and throw a useful exception if `max_docs=auto` is configured? (as opposed to a number parse exception)

Reply:
Nice idea, on it.

Reply:
I implemented this but in order to do it properly I had to expand the scope to apply both in the setting parsing and the XContent. Let me know what you think, personally, I think this is the right approach and this way the process to turn a condition to `auto` is the same no matter if it's a `Long` or a `ByteSizeValue`.